src/modules/Resources/devices.types.ts

```diff
@@ -91,52 +91,65 @@ interface DeviceCreateInfoLegacy extends Omit<DeviceCreateInfoBasic, "type"> {
 type DeviceCreateInfo = DeviceCreateInfoLegacy | DeviceCreateInfoMutable | DeviceCreateInfoImmutable;
 type DeviceEditInfo = Partial<Omit<DeviceCreateInfo, "chunk_period" | "type"> & { chunk_retention: number }>;
 
-interface DeviceInfo extends Required<Omit<DeviceCreateInfoBasic, "configuration_params">> {
+interface DeviceInfoBase {
   /**
    * Device ID.
    */
   id: GenericID;
-  /**
-   * Device's data storage (bucket) type.
-   */
-  type: DataStorageType;
```

Comment:
This is duplicated from the type union on `DeviceInfo` merging both `"mutable"` and `"immutable"`